pyOCD/gdbserver/gdbserver.py

```diff
@@ -329,12 +338,27 @@ def run(self):
                     break
 
                 if self.packet_io.interrupt_event.isSet():
-                    logging.debug("Got unexpected ctrl-c, ignoring")
+                    if self.non_stop:
```

Comment:
Probably not. In all-stop mode, the only time the target should be running is in `resume()`, where we already have a loop that checks for ctrl-c.

But now I'm wondering if all-stop should be more integrated into the main packet receive loop, like non-stop is. `resume()` would just resume the target and set the target-running flag. Then the main loop would check for ctrl-c as with non-stop. Originally, I was trying to add non-stop with as little change to the rest of the code as possible, but maybe that's not the right way to do it. Or maybe we just clean this up later.